SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -21,62 +21,113 @@ SpatialLoadBalanceEnforcer::SpatialLoadBalanceEnforcer(const PhysicalWorkerName&
 
 void SpatialLoadBalanceEnforcer::OnAuthorityIntentComponentUpdated(const Worker_ComponentUpdateOp& Op)
 {
-	check(Op.update.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID)
-	if (StaticComponentView->HasAuthority(Op.entity_id, SpatialConstants::ENTITY_ACL_COMPONENT_ID))
+	check(Op.update.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID);
+	if (CanEnforce(Op.entity_id))
 	{
-		QueueAclAssignmentRequest(Op.entity_id);
+		MaybeQueueAclAssignmentRequest(Op.entity_id);
```

Comment:
Instead of wrapping every call with `CanEnforce`, just do it internally.

Reply:
genius, done